src/cliargsparser.cpp

```diff
@@ -7,265 +7,181 @@
-
-	char* rs_cliargsparser_readinfo_import_file(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_do_read_export(void* rs_cliargsparser);
-
-	char* rs_cliargsparser_readinfo_export_file(void* rs_cliargsparser);
-
-	unsigned int rs_cliargsparser_show_version(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_silent(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_using_nonstandard_configs(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_should_return(void* rs_cliargsparser);
-
-	int rs_cliargsparser_return_code(void* rs_cliargsparser);
-
-	char* rs_cliargsparser_display_msg(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_should_print_usage(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_refresh_on_start(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_set_url_file(void* rs_cliargsparser);
-
-	char* rs_cliargsparser_url_file(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_set_lock_file(void* rs_cliargsparser);
-
-	char* rs_cliargsparser_lock_file(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_set_cache_file(void* rs_cliargsparser);
-
-	char* rs_cliargsparser_cache_file(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_set_config_file(void* rs_cliargsparser);
-
-	char* rs_cliargsparser_config_file(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_execute_cmds(void* rs_cliargsparser);
-
-	unsigned int rs_cliargsparser_cmds_to_execute_count(void* rs_cliargsparser);
-	char* rs_cliargsparser_cmd_to_execute_n(void* rs_cliargsparser, unsigned int n);
-
-	bool rs_cliargsparser_set_log_file(void* rs_cliargsparser);
-
-	char* rs_cliargsparser_log_file(void* rs_cliargsparser);
-
-	bool rs_cliargsparser_set_log_level(void* rs_cliargsparser);
-
-	char rs_cliargsparser_log_level(void* rs_cliargsparser);
-}
-
-#define GET_VALUE(NAME, DEFAULT) \
-	if (rs_cliargsparser) { \
-		return rs_cliargsparser_ ## NAME (rs_cliargsparser); \
-	} else { \
-		return DEFAULT; \
-	}
-
-#define GET_STRING(NAME) \
-	if (rs_cliargsparser) { \
-		return RustString(rs_cliargsparser_ ## NAME (rs_cliargsparser)); \
-	} else { \
-		return {}; \
-	}
-
-#define GET_OPTIONAL_VALUE(CHECKER, GETTER, DEFAULT) \
-	if (rs_cliargsparser) { \
-		if (rs_cliargsparser_ ## CHECKER (rs_cliargsparser)) { \
-			return rs_cliargsparser_ ## GETTER (rs_cliargsparser); \
-		} else { \
-			return nonstd::nullopt; \
-		} \
-	} else { \
-		return DEFAULT; \
-	}
-
-#define GET_OPTIONAL_STRING(CHECKER, GETTER) \
-	if (rs_cliargsparser) { \
-		if (rs_cliargsparser_ ## CHECKER (rs_cliargsparser)) { \
-			return RustString(rs_cliargsparser_ ## GETTER (rs_cliargsparser)); \
-		} else { \
-			return nonstd::nullopt; \
-		} \
-	} else { \
-		return {}; \
-	}
-
 namespace newsboat {
 
-CliArgsParser::CliArgsParser(int argc, char* argv[])
+
+rust::Vec<rust::String> argv_to_rust_args(int argc, char* argv[])
 {
-	rs_cliargsparser = create_rs_cliargsparser(argc, argv);
+	rust::Vec<rust::String> args;
+	for (int i = 0; i < argc; ++i) {
+		// TODO: Handle invalid utf-8 codepoints gracefully?
```

Comment:
Oh, looks like you've stumbled upon a design flaw in the Rust implementation of `CliArgsParser`: it represents `argv` as `Vec<String>` rather than `Vec<OsString>`. The difference is that `String` has a known encoding (UTF-8), whereas `OsString` is just a bunch of bytes. The latter is a "more correct" type, because we really don't know what the user passed us on the command line—it could be a filepath, which on Unix doesn't have a guaranteed encoding (all we know is that it's an array of bytes that don't include `NUL` or `/`).

I think we *could* print out an error here and abort the program, but that's just papering over a problem. A correct solution would be to switch `CliArgsParser` to `OsString` — I'll file an issue about that. Please remove this TODO.